checks/alertmanager.go

```diff
@@ -88,3 +89,25 @@ func extractMessage(annotations map[string]string) string {
 	}
 	return ""
 }
+
+func generateNameSuffix(name string, labels map[string]string) string {
+	// Based on the type of alert, we will add alert metadata to the check name
+	// The key here is the alert type and the name will be "/".join(value)
+	nameSuffixFromLabels := map[string][]string{
+		"KubePodCrashLooping":            {"namespace", "pod"},
+		"KubePodNotReady":                {"namespace", "pod"},
+		"KubeDeploymentReplicasMismatch": {"namespace", "deployment"},
```

Comment:
Can we make the heuristic and not depend on the alert name? Also need to account for Daemonset and Statefulset